ignite/pkg/gomodulepath/gomodulepath.go

```diff
@@ -73,13 +79,65 @@ func Find(path string) (parsed Path, appPath string, err error) {
 	return Path{}, "", errors.Wrap(gomodule.ErrGoModNotFound, "could not locate your app's root dir")
 }
 
-func validateModulePath(path string) error {
+// ExtractUserRepoNames extracts the user and repository names from a go module path.
+func ExtractUserRepoNames(path string) (user string, repo string, err error) {
+	if path == "" {
+		return "", "", errors.New("module path is empty")
+	}
+
+	items := strings.Split(path, "/")
+
+	// Remove the first path item if it is assumed to be a domain name
+	if len(items) > 1 && strings.Contains(items[0], ".") {
+		items = items[1:]
+	}
+
+	count := len(items)
+	if count == 1 {
+		// The path is a single name that its used as user and repository name
```

Comment:
Should we really assume that user and repo names are the same (because in reality, that's not relevant. 

Module name can be a single word in the go.mod. I believe we shouldn't treat module names as Git compatible urls anymore. Because they can be and cannot be.

We should even rename the func name `ExtractUserRepoNames` to something more generic.

Wdyt?

Reply:
With the latest changes "user", "owner" and "repo" were replaced by "app module paths".

The `ExtractUserRepoNames` is now `ExtractAppPath` and what it does is extracting the app module path from a Go module path:
```go
// "github.com/username/repo" => "username/repo"
// "a/b/c/namespace/name" => "namespace/name"
// "name" => "name"
ExtractAppPath(path string) string
```

Also it could be renamed to `ExtractAppModulePath` to be more specific.